rules.py

```diff
@@ -680,6 +680,31 @@ def check(self, element):
                 self.labels.add(element_label)
 
 
+class ReusedCandidate(base.BaseRule):
+    """Candidate should be referred to by only one contest.
+
+    A Candidate object should only ever be referenced from one contest. If a Person
+    is running in multiple Contests, then that Person is a Candidate several times over,
+    but a Candida(te|cy) can't span contests.
+    """
+    seen_candidates = []
+
+    def elements(self):
+        return ["CandidateSelection"]
+
+    def check(self, element):
+        candidate_ids  = element.find("CandidateIds")
+        if candidate_ids is None:
+          return
+        for candidate_id in candidate_ids.text.split():
+            if candidate_id in self.seen_candidates:
+                raise base.ElectionError(
+                    "Line %d. Candidate %s is already associated with another contest"
```

Comment:
In order to provide a better error message to the user, I'd recommend keeping a dictionary that maps candidate IDs to the list of contest ID(s) in which they're referenced. The check would iterate over every contest, appending the contest ID to the list for each candidate referenced by that contest. At the very end, every candidate who has a contest list of len > 1, print an error message saying that the candidate is referenced by contests X, Y, and Z.